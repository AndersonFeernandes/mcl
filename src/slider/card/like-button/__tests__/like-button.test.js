import "./../../../../test.setup";

import React from "react";
import { shallow, mount } from "enzyme";
import LikeButton from "../like-button";

describe("Testing LikeButton Component", () => {

    it("should render correctly when is liked", (done) => {
        const id = 1, isLiked = true;
        const wrapper = shallow(
            <LikeButton id={id} isLiked={isLiked} />
        );

        expect(wrapper).toMatchSnapshot();
        done();
    });

    it("should render correctly when is unliked", (done) => {
        const id = 1, isLiked = false;
        const wrapper = shallow(
            <LikeButton id={id} isLiked={isLiked} />
        );

        expect(wrapper).toMatchSnapshot();
        done();
    });

    describe("Testing click function", () => {

        beforeEach(() => {
            sinon.restore();
        });

        it("if click on like button when its liked, it should unlike", (done) => {
            const id = 1, isLiked = true;
            const mockData = { is_liked: !isLiked };
            const promise = Promise.resolve(new Response(JSON.stringify(mockData)));
            const stub = sinon.stub(global, "fetch").callsFake(() => promise);

            const wrapper = mount(
                <LikeButton id={id} isLiked={isLiked} />
            );

            wrapper.first().simulate("click");
            setImmediate(() => {
                expect(wrapper.state()).toHaveProperty("isLiked", !isLiked);
                done();
            });

        });

        it("if click on like button when its unliked, it should like", (done) => {
            const id = 1, isLiked = false;
            const mockData = { is_liked: !isLiked };
            const promise = Promise.resolve(new Response(JSON.stringify(mockData)));
            const stub = sinon.stub(global, "fetch").callsFake(() => promise);

            const wrapper = mount(
                <LikeButton id={id} isLiked={isLiked} />
            );

            wrapper.first().simulate("click");
            setImmediate(() => {
                expect(wrapper.state()).toHaveProperty("isLiked", !isLiked);
                done();
            });

        });
    });


});