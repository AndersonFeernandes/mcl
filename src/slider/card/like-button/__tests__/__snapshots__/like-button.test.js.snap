// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LikeButton Component should render correctly when is liked 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LikeButton
    id={1}
    isLiked={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "like",
      "onClick": [Function],
      "onKeyDown": [Function],
      "tabIndex": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "like",
        "onClick": [Function],
        "onKeyDown": [Function],
        "tabIndex": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Testing LikeButton Component should render correctly when is unliked 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LikeButton
    id={1}
    isLiked={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "unlike",
      "onClick": [Function],
      "onKeyDown": [Function],
      "tabIndex": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "unlike",
        "onClick": [Function],
        "onKeyDown": [Function],
        "tabIndex": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
